api/policies/v1/admissionpolicy_webhook_test.go

```diff
@@ -16,77 +14,39 @@ limitations under the License.
 package v1
 
 import (
+	"context"
 	"testing"
 
+	"github.com/go-logr/logr"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
 	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
 
 	"github.com/kubewarden/kubewarden-controller/internal/constants"
 )
 
 func TestAdmissionPolicyDefault(t *testing.T) {
-	policy := AdmissionPolicy{}
-	policy.Default()
+	defaulter := admissionPolicyDefaulter{logger: logr.Discard()}
+	policy := &AdmissionPolicy{}
 
-	require.Equal(t, constants.DefaultPolicyServer, policy.GetPolicyServer())
-	require.Contains(t, policy.GetFinalizers(), constants.KubewardenFinalizer)
+	err := defaulter.Default(context.Background(), policy)
+	require.NoError(t, err)
+
+	assert.Equal(t, constants.DefaultPolicyServer, policy.GetPolicyServer())
+	assert.Contains(t, policy.GetFinalizers(), constants.KubewardenFinalizer)
 }
 
 func TestAdmissionPolicyValidateCreate(t *testing.T) {
+	validator := admissionPolicyValidator{logger: logr.Discard()}
 	policy := NewAdmissionPolicyFactory().Build()
-	warnings, err := policy.ValidateCreate()
-	require.NoError(t, err)
-	require.Empty(t, warnings)
-}
 
-func TestAdmissionPolicyValidateUpdate(t *testing.T) {
-	oldPolicy := NewAdmissionPolicyFactory().Build()
-	newPolicy := NewAdmissionPolicyFactory().Build()
-	warnings, err := newPolicy.ValidateUpdate(oldPolicy)
+	warnings, err := validator.ValidateCreate(context.Background(), policy)
 	require.NoError(t, err)
-	require.Empty(t, warnings)
-
-	oldPolicy = NewAdmissionPolicyFactory().
-		WithMode("monitor").
-		Build()
-	newPolicy = NewAdmissionPolicyFactory().
-		WithMode("protect").
-		Build()
-	warnings, err = newPolicy.ValidateUpdate(oldPolicy)
-	require.NoError(t, err)
-	require.Empty(t, warnings)
+	assert.Empty(t, warnings)
 }
 
-func TestInvalidAdmissionPolicyValidateUpdate(t *testing.T) {
-	oldPolicy := NewAdmissionPolicyFactory().
-		WithPolicyServer("old").
-		Build()
-	newPolicy := NewAdmissionPolicyFactory().
-		WithPolicyServer("new").
-		Build()
-	warnings, err := newPolicy.ValidateUpdate(oldPolicy)
-	require.Error(t, err)
-	require.Empty(t, warnings)
-
-	newPolicy = NewAdmissionPolicyFactory().
-		WithPolicyServer("new").
-		WithMode("monitor").
-		Build()
-
-	warnings, err = newPolicy.ValidateUpdate(oldPolicy)
-	require.Error(t, err)
-	require.Empty(t, warnings)
-}
-
-func TestAdmissionPolicyValidateUpdateWithInvalidOldPolicy(t *testing.T) {
```

Comment:
This and similar tests in other files are unnecessary since the API server will never send incorrect objects.

Reply:
good point

Reply:
added missing invalid types tests, added ValidateDelete tests, added missing PolicyServer webhook tests.